src/tests/cptests/cptests.cc

```diff
@@ -951,6 +1012,52 @@ void cptest_servicestatus()
     delete cc;
 }
 
+void cptest_sendsignal()
+{
+    using namespace std;
+
+    service_set sset;
+    ha_string command = "test-command";
+    list<pair<unsigned,unsigned>> command_offsets;
+    command_offsets.emplace_back(0, command.length());
+    std::list<prelim_dep> depends;
+
+    process_service p {&sset, "test-service", std::move(command), command_offsets, depends};
+    init_service_defaults(p);
+    sset.add_service(&p);
+
+    p.start();
+    sset.process_queues();
+    base_process_service_test::exec_succeeded(&p);
+    sset.process_queues();
+
+    int fd = bp_sys::allocfd();
+    auto *cc = new control_conn_t(event_loop, &sset, fd);
+
+    // Get a service handle:
+    control_conn_t::handle_t h = find_service(fd, "test-service", service_state_t::STARTED,
+            service_state_t::STARTED);
+
+    // Issue a signal (SIGHUP for example):
+    std::vector<char> cmd = { DINIT_CP_SIGNAL, SIGHUP };
```

Comment:
Hmm, I think I missing something but `SIGHUP` doesn't allocate 4 bytes which expected in `control.cc`. How it should be?

(Sorry I know I should fix anything related to this PR by myself but I think a simple question is reasonable)

Reply:
Since it's a vector of `char` the initialiser list is interpreted as a a list of `char` (1 byte each). The first `char` is `DINIT_CP_SIGNAL`, the next will be `SIGHUP`. To store `SIGHUP` as a sequence of four bytes you need to use the same technique as is used for a `handle_t` (which is also a four-byte integer) on the following lines. I.e. put it in a `int32_t` variable, copy each byte from the value into the vector.

Reply:
How is it that you're able to push commits to helican's repo btw?

Reply:
> How is it that you're able to push commits to helican's repo btw?

Because he enabled "Maintainers can edit this PR directly" which allow direct pushing for anyone have write access to main repo. (But only for PR's branch)

It's very useful thing for some situations like this.